Adopt kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence

jyavenard opened the pull request:
#### 144b9069f26b3503193252e30181c269fe09d916
<pre>
Adopt kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence
<a href="https://bugs.webkit.org/show_bug.cgi?id=242681">https://bugs.webkit.org/show_bug.cgi?id=242681</a>
&lt;rdar://83910012&gt;

Reviewed by Dean Jackson.

When demuxing contiguous data in a webm, we know that there shouldn&apos;t be any
discontinuity between samples, however the container may wrongly indicates that
there&apos;s such gap.
We know tag each sample block with the key kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence
where we know there shouldn&apos;t be a discontinuity and make the audio layer
not insert silences.

This change is currently non-functional has kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence
isn&apos;t properly behaving correctly until rdar://87997713 is resolved.

* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.cpp:
* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.h:
* Source/WebCore/platform/MediaSample.h:
(WebCore::MediaSamplesBlock::discontinuity const):
(WebCore::MediaSamplesBlock::setDiscontinuity):
* Source/WebCore/platform/graphics/cocoa/CMUtilities.mm:
(WebCore::toCMSampleBuffer):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::SourceBufferParserWebM::parsedMediaData):
(WebCore::SourceBufferParserWebM::flushPendingAudioSamples):
(WebCore::SourceBufferParserWebM::appendData):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.h:

Canonical link: <a href="https://commits.webkit.org/252561@main">https://commits.webkit.org/252561@main</a>
</pre>

grorg commented on Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
```diff
@@ -1458,6 +1459,8 @@ void SourceBufferParserWebM::flushPendingAudioSamples()
         return;
     ASSERT(m_audioInfo);
     m_queuedAudioSamples.setInfo(m_audioInfo.copyRef());
+    m_queuedAudioSamples.setDiscontinuity(m_audioDiscontinuity);
```
Is this ok if `m_audioDiscontinuity` is `nullopt`? Can it ever get to that point?

Duh, ignore that - I was looking at `m_discontinuity`.

jyavenard committed:
Adopt kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence
https://bugs.webkit.org/show_bug.cgi?id=242681
<rdar://83910012>

Reviewed by Dean Jackson.

When demuxing contiguous data in a webm, we know that there shouldn't be any
discontinuity between samples, however the container may wrongly indicates that
there's such gap.
We know tag each sample block with the key kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence
where we know there shouldn't be a discontinuity and make the audio layer
not insert silences.

This change is currently non-functional has kCMSampleBufferAttachmentKey_FillDiscontinuitiesWithSilence
isn't properly behaving correctly until rdar://87997713 is resolved.

* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.cpp:
* Source/WebCore/PAL/pal/cf/CoreMediaSoftLink.h:
* Source/WebCore/platform/MediaSample.h:
(WebCore::MediaSamplesBlock::discontinuity const):
(WebCore::MediaSamplesBlock::setDiscontinuity):
* Source/WebCore/platform/graphics/cocoa/CMUtilities.mm:
(WebCore::toCMSampleBuffer):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::SourceBufferParserWebM::parsedMediaData):
(WebCore::SourceBufferParserWebM::flushPendingAudioSamples):
(WebCore::SourceBufferParserWebM::appendData):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.h:

Canonical link: https://commits.webkit.org/252561@main
